core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -87,13 +89,23 @@ class SnapshotService(protected val userInfo: UserInfo, val dataSource: SlickDat
     }
   }
 
-  def enumerateSnapshots(workspaceName: WorkspaceName, offset: Int, limit: Int): Future[ResourceList] = {
+  def massageSnapshots(references: ResourceList): Seq[DataRepoSnapshotResource] = {
+    val res = new ListBuffer[DataRepoSnapshotResource]()
+    references.getResources.forEach(r => { val massaged = new DataRepoSnapshotResource
+                                          massaged.setAttributes(r.getResourceAttributes.getGcpDataRepoSnapshot)
+                                          massaged.setMetadata(r.getMetadata)
+                                          res += massaged
+    } )
+    res
+  }
+
+  def enumerateSnapshots(workspaceName: WorkspaceName, offset: Int, limit: Int): Future[Seq[DataRepoSnapshotResource]] = {
     getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.read, Some(WorkspaceAttributeSpecs(all = false))).map { workspaceContext =>
       Try(workspaceManagerDAO.enumerateDataRepoSnapshotReferences(workspaceContext.workspaceIdAsUUID, offset, limit, userInfo.accessToken)) match {
-        case Success(references) => references
+        case Success(references) => massageSnapshots(references)
         // if we fail with a 404, it means we have no stub in WSM yet. This is benign and functionally equivalent
         // to having no references, so return the empty list.
-        case Failure(ex: bio.terra.workspace.client.ApiException) if ex.getCode == 404 => new ResourceList()
+        case Failure(ex: bio.terra.workspace.client.ApiException) if ex.getCode == 404 => List()
```

Comment:
open for debate: I prefer to be explicit about types, so I'd use `Seq.empty[DataRepoSnapshotResource]` instead of `List()`; you could also use `Seq()`. Some prefer the terseness/readability of the shorter `Seq()` but I prefer to see it written out. Scala will do the right thing in either case.